sycl/source/detail/graph_impl.cpp

```diff
@@ -226,6 +360,85 @@ bool graph_impl::clearQueues() {
   return AnyQueuesCleared;
 }
 
+void graph_impl::printGraphAsDot(const char *GraphName) {
+  /// Vector of nodes visited during the graph printing
+  std::vector<std::shared_ptr<node_impl>> MVisitedNodes;
+  std::string GraphNameStr(GraphName);
+  std::string FileName = "graph_" + GraphNameStr + ".dot";
+
+  std::fstream Stream(FileName, std::ios::out);
+  Stream << "strict digraph {" << std::endl;
+
+  for (std::shared_ptr<node_impl> Node : MRoots)
+    printDotRecursive(Stream, MVisitedNodes, Node);
+
+  Stream << "}" << std::endl;
+
+  unmarkVisitedNodes(MVisitedNodes);
+}
+
+bool graph_impl::hasSimilarStructure(std::shared_ptr<detail::graph_impl> Graph,
+                                     bool DebugPrint) const {
+  if (this == Graph.get())
+    return true;
+
+  if (MContext != Graph->MContext) {
+    if (DebugPrint)
+      std::cout << "MContext are not the same." << std::endl;
+    return false;
+  }
+
+  if (MDevice != Graph->MDevice) {
+    if (DebugPrint)
+      std::cout << "MDevice are not the same." << std::endl;
+    return false;
+  }
+
+  if (MRecordingQueues.size() != Graph->MRecordingQueues.size()) {
+    if (DebugPrint)
+      std::cout << "MRecordingQueues sizes are not the same." << std::endl;
+    return false;
+  }
```

Comment:
Should the recording queues be included in this comparison? I don't think they contribute to the structure of the graph.

Reply:
I agree.